packages/api/tests/mock.factories.js

```diff
@@ -1142,32 +1142,80 @@ function fakeProduct(defaultData = {}) {
   };
 }
 
+async function soil_amendment_methodFactory() {
+  return knex('soil_amendment_method').insert({ key: faker.lorem.word() }).returning('*');
+}
+
+async function soil_amendment_purposeFactory() {
+  return knex('soil_amendment_purpose').insert({ key: faker.lorem.word() }).returning('*');
+}
+
 async function soil_amendment_taskFactory(
-  { promisedTask = taskFactory(), promisedProduct = productFactory() } = {},
+  {
+    promisedTask = taskFactory(),
+    promisedProduct = productFactory(),
+    promisedMethod = soil_amendment_methodFactory(),
+  } = {},
   soil_amendment_task = fakeSoilAmendmentTask(),
 ) {
-  const [task, product] = await Promise.all([promisedTask, promisedProduct]);
+  const [task, product, method] = await Promise.all([
+    promisedTask,
+    promisedProduct,
+    promisedMethod,
+  ]);
   const [{ task_id }] = task;
-  const [{ product_id }] = product;
+  const [{ method_id }] = method;
   return knex('soil_amendment_task')
-    .insert({ task_id, product_id, ...soil_amendment_task })
+    .insert({
+      task_id,
+      method_id,
+      ...soil_amendment_task,
+    })
     .returning('*');
 }
 
 function fakeSoilAmendmentTask(defaultData = {}) {
   return {
-    product_quantity: faker.datatype.number(),
-    purpose: faker.helpers.arrayElement([
-      'structure',
-      'moisture_retention',
-      'nutrient_availability',
-      'ph',
-      'other',
-    ]),
     ...defaultData,
   };
 }
 
+async function soil_amendment_task_productsFactory(
+  { promisedSoilAmendmentTask = soil_amendment_taskFactory() } = {},
+  soil_amendment_task_product = fakeSoilAmendmentTaskProduct(),
+) {
+  const [soilAmendmentTask] = await promisedSoilAmendmentTask;
+  return knex('soil_amendment_task_products')
+    .insert({
+      task_id: soilAmendmentTask.task_id,
+      ...soil_amendment_task_product,
+    })
+    .returning('*');
+}
+
+function fakeSoilAmendmentTaskProduct() {
```

Comment:
I think this function should also take `defaultData = {}` so we can do:
```
mocks.fakeSoilAmendmentTaskProduct({ product_id })
```
instead of
```
{ product_id, ...mocks.fakeSoilAmendmentTaskProduct() }
```

Reply:
Thank you, I feel like I should have noticed how awkwardly I wrote that 😅